worker-document/src/main/java/com/hpe/caf/worker/document/impl/ScriptImpl.java

```diff
@@ -0,0 +1,265 @@
+
+    private int lastKnownIndex;
+    private String name;
+    private AbstractScriptSpec scriptSpec;
+    private boolean isInstalled;
+    private Bindings loadedScriptBindings;
+
+    public ScriptImpl(
+        final ApplicationImpl application,
+        final AbstractTask task
+    )
+    {
+        super(application);
+        this.task = Objects.requireNonNull(task);
+        this.lastKnownIndex = 0;
+        this.name = null;
+        this.scriptSpec = null;
+        this.isInstalled = true;
+        this.loadedScriptBindings = null;
+    }
+
+    public ScriptImpl(
+        final ApplicationImpl application,
+        final AbstractTask task,
+        final DocumentWorkerScript script
+    ) throws InvalidScriptException
+    {
+        this(application, task);
+        this.name = Objects.requireNonNull(script).name;
+        this.scriptSpec = AbstractScriptSpec.create(application, script);
+    }
+
+    @Override
+    public int getIndex()
+    {
+        final int currentIndex = task.getScripts().getCurrentScriptIndex(lastKnownIndex, this);
+
+        lastKnownIndex = currentIndex;
+        return currentIndex;
+    }
+
+    @Override
+    public String getName()
+    {
+        return name;
+    }
+
+    @Override
+    public Object getObject()
+    {
+        final Bindings bindings = loadedScriptBindings;
+
+        return (bindings == null)
+            ? null
+            : bindings.get("thisScriptObject");
+    }
+
+    @Override
+    public String getScript() throws IOException
+    {
+        return (scriptSpec == null)
+            ? null
+            : scriptSpec.getScript();
+    }
+
+    @Nonnull
+    @Override
+    public Task getTask()
+    {
+        return task;
+    }
+
+    @Override
+    public void install()
+    {
+        isInstalled = true;
+    }
+
+    @Override
+    public boolean isInstalled()
+    {
+        return isInstalled;
+    }
+
+    @Override
+    public boolean isLoaded()
+    {
+        return (loadedScriptBindings != null);
+    }
+
+    @Override
+    public void load() throws ScriptException
+    {
+        // Check that the script isn't already loaded
+        if (isLoaded()) {
+            return;
+        }
+
+        // Check that the script has been initialised
+        if (scriptSpec == null) {
+            throw new RuntimeException("The script must be initialized before it can be loaded.");
+        }
+
+        // Get a reference to the script manager
+        final JavaScriptManager javaScriptManager = application.getJavaScriptManager();
+
+        // Get the object code for the script
+        final CompiledScript scriptObjectCode = javaScriptManager.getObjectCode(name, scriptSpec);
+
+        // Create a new global context for the script to run in
+        final Bindings newGlobal = javaScriptManager.createNewGlobal();
+
+        // Add a reference to this script object into the script's global context
+        newGlobal.put("thisScript", this);
```

Comment:
No - the `thisScriptObject` is one that the script can set to expose something.  This global variable has a different purpose - it is to allow the script to access itself, in case it needs to unload or uninstall itself for example.